pkg/controller/core/dynamicresourceallocationconfig_controller.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/features"
+)
+
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=dynamicresourceallocationconfigs,verbs=get;list;watch
+
+// DynamicResourceAllocationConfigReconciler keeps the cache's DeviceClass→logical-resource
+// mapping in sync with the singleton DynamicResourceAllocationConfig object.
+// It is only registered when the DynamicResourceAllocation feature gate is enabled.
+type DynamicResourceAllocationConfigReconciler struct {
+	client    client.Client
+	cache     *cache.Cache
+	log       logr.Logger
+	namespace string
+}
+
+func NewDRAConfigReconciler(client client.Client, c *cache.Cache, namespace string) *DynamicResourceAllocationConfigReconciler {
+	return &DynamicResourceAllocationConfigReconciler{
+		client:    client,
+		cache:     c,
+		log:       ctrl.Log.WithName("dra-config-reconciler"),
+		namespace: namespace,
+	}
+}
+
+func (r *DynamicResourceAllocationConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if !features.Enabled(features.DynamicResourceAllocation) {
+		return ctrl.Result{}, nil
+	}
+
+	var cfg kueuealpha.DynamicResourceAllocationConfig
+	if err := r.client.Get(ctx, req.NamespacedName, &cfg); err != nil {
+		// TODO: this should never happen. But if a user deletes the singleton DRA by mistake, we should
```

Comment:
I don't agree with this comment. Admin can create CRDs and/or remove them at will. If we build and publish this CRD at install time then we could assume that this does not happen.

But we should expect cases where this object does not exist and we should handle if its deleted.

Reply:
We use finalizers for CQ and RF.

If a workload or CQ is using a RF, we set a finalizer.

Reply:
Maybe we should add a finalizer if any CQ references DRA?

Reply:
I'm also okay to file this as issue for a future work.

But what would be the behavior if someone deletes this and the CQ references it?